libs/dotnet-sdk/FeatureBoardHttpClient.cs

```diff
@@ -9,23 +8,18 @@ namespace FeatureBoard.DotnetSdk;
 internal class FeatureBoardHttpClient : IFeatureBoardHttpClient
 {
   private readonly HttpClient _httpClient;
-  private readonly ILogger<FeatureBoardHttpClient> _logger;
+  private readonly ILogger _logger;
 
-  public FeatureBoardHttpClient(HttpClient httpClient, IOptions<FeatureBoardOptions> options, ILogger<FeatureBoardHttpClient> logger)
+  public FeatureBoardHttpClient(HttpClient httpClient, ILogger<FeatureBoardHttpClient> logger)
   {
     _httpClient = httpClient;
-    _httpClient.BaseAddress = options.Value.HttpEndpoint;
```

Comment:
I'd prefer these to stay with the client so if someone breaks away from using register feature board, this they don't have to know to do this.

Reply:
I hear where you are coming from. However if they're not using the `RegisterFeatureBoard` extension then there is also the risk that consumers are passing in a _shared_ `HttpClient` (eg. they could just use the basic [`AddHttpClient()` extension](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.httpclientfactoryservicecollectionextensions.addhttpclient?view=dotnet-plat-ext-7.0#microsoft-extensions-dependencyinjection-httpclientfactoryservicecollectionextensions-addhttpclient(microsoft-extensions-dependencyinjection-iservicecollection))) to `FeatureBoardHttpClient`. This would be fine so long as they don't attempt to utilise that shared `HttpClient` elsewhere in their application, but if they do the code [removed here] will be setting defaults on that common shared instance, potentially causing weird somewhat hard to reproduce bugs.

I'll think on this further, see if I can come up with something which addresses both concerns.

Reply:
Dug into this a bit more and it is effectively impossible for consumers to "new up" a `FeatureBoardHttpClient` independantly as you've declared that class with `internal` visibility, The only way they can [legitimately] obtain an instance of that class is by using `RegisterFeatureBoard.AddFeatureBoard(...)` and declaring dependency on it in a consuming class. Hence the concern about keeping `HttpClient` configuration self contained within `FeatureBoardHttpClient` moot.

At the risk of being dogmatic, I'd recommend keeping the config code as I have it because it reduces the number of `FeatureBoardHttpClient`'s collaborators and better follows the [Constructor Injection design pattern](https://blog.ploeh.dk/2011/03/03/InjectionConstructorsshouldbesimple/).